staging/src/k8s.io/apiserver/pkg/admission/interfaces.go

```diff
@@ -62,6 +62,20 @@ type Attributes interface {
 	AddAnnotation(key, value string) error
 }
 
+// ObjectInterfaces is an interface used by AdmissionController to get object interfaces
+// such as Converter or Defaulter. These interfaces are normally coming from Request Scope
+// to handle special cases like CRDs.
+type ObjectInterfaces interface {
```

Comment:
@mbohlool @liggitt adding this as a parameter to all admission controller would widely change all the admissions' interface. how about injecting a dynamic scheme getter func from the initializers instead a fixed scheme? sth like `func GetScheme(gvk)`, if the gvk's not registered in the legacy scheme(which is, the requesting resource is not standard), it returns the extension scheme.

Reply:
> how about injecting a dynamic scheme getter func from the initializers instead a fixed scheme? sth like `func GetScheme(gvk)`, if the gvk's not registered in the legacy scheme(which is, the requesting resource is not standard), it returns the extension scheme.

That seems more complex and error-prone. The choice is not between the legacy scheme and the extensions scheme, but between the legacy scheme and a per-resource converter/typer/defaulter for each custom resource. The object interfaces for the object being handled are in the rest handler, and fit much more naturally as a parameter to admit/validate.